src/applications/disability-benefits/all-claims/utils/on-submit/purge-toxic-exposure-data.js

```diff
@@ -92,6 +93,12 @@ const purgeExposureDetails = (toxicExposure, exposureType, mapping) => {
     delete result[detailsKey];
   }
 
+  // Remove orphaned otherKey when parent exposure doesn't exist
+  const hasOrphanedOtherKey = otherKey && result[otherKey] && !hasExposure;
+  if (hasOrphanedOtherKey) {
+    delete result[otherKey];
+  }
```

Comment:
Although I think the result of this logic is as I would expect, this is a bit confusing for me because it reads as though the "parent exposure" for `otherHerbicideLocations` would be `herbicide`, even though it is actually `toxicExposure`.